app/controllers/users_controller.rb

```diff
@@ -14,15 +24,35 @@ def new
 
   def create
     user = User.new(user_params)
-    
+
     authenticate!(:password) if user.save
+    unless params[:invitation_token].blank?
+      @website_member = WebsiteMember.find_by_invitation_token(params[:invitation_token])
+      unless @website_member.nil?
+        @website_member.update_attributes(user_id: user.id, role: 2)
+        set_website(Website.find(@website_member.website_id))
+      end
```

Comment:
this code should be in the user model and call this by doing `user.accept(params[:invitation_token])`